### ISSUE TITLE
[css-scroll-snap-2] scroll-start-target: auto doesn't match general meaning of auto

### ISSUE THREAD

**Issue by DavMila** [w3c/csswg-drafts#11173]
The `scroll-start-target` [spec](https://drafts.csswg.org/css-scroll-snap-2/#scroll-start-target) specifies 2 keywords for the `scroll-start-target` property:
* `none`, meaning: this element is not to be used to  determine the initial scroll position  of its nearest scroll container, and
* `auto`, meaning: this element should be used to determine the initial scroll position of its nearest scroll container.  

The use of `auto` in this way is a bit of a departure from what `auto` means in CSS, i.e. "the UA should do what it thinks is best."

Should we change `auto` to a different keyword? Here are a few we could consider: `on-first-appearance` or `align` or `active` or `visible`. My preference would be `align`.

**Comment by DavMila** [comment]
FYI @flackr

**Comment by argyleink** [comment]
time to bikeshed! lol. I do think a different value is better.

options?
1. first-paint
2. page-load
3. me
4. scroll-ready
5. loaded
6. on-first-appearance
7. align
8. active
9. visible
10. init
11. here

**Comment by flackr** [comment]
Since we have ideas that we might try to include all scroll-start-targets, we could consider something like `include`?

Or, we could consider other values we may want to eventually support. E.g. if we wanted to eventually have options that align with / explain [scrolling to a fragment](https://html.spec.whatwg.org/#scrolling-to-a-fragment), we could consider what the names might be if we want to describe the ways that this is currently different from that and/or if we eventually wanted to be able to specify a scroll that was roughly equivalent to scrolling to a fragment.

In particular, scrolling to a fragment:
* Scrolls all ancestor scrollers as well, e.g. `scroll-start-target: none | local | global`?
* Only scrolls if discovered during the navigation, but not if you later add the node to the document, e.g. `scroll-start-target: none | navigation | always`

**Comment by DavMila** [comment]
It's also been [pointed out](https://github.com/w3ctag/design-reviews/issues/1011#issuecomment-2471818871) that within CSS it might be very common to assosciate`start` with the edges of a box, e.g. "start" occurs in `margin-block-start`, `inset-block-start`, `border-block-start`, `padding-block-start`, etc., and could therefore be confusing in `scroll-start-target`. Perhaps  we we could give the property a better name.
Some suggestions:
- `scroll-origin-anchor`
- `scroll-origin-target`

It probably makes sense to talk about the name of the property before talking about what keywords are appropriate.

**Comment by argyleink** [comment]
i like `scroll-origin` 👍🏻

**Comment by johannesodland** [comment]
Wouldn't `scroll-origin-*` possibly be confusing, as we're not affecting the [scroll origin](https://www.w3.org/TR/css-overflow-3/#scroll-origin), but the initial scroll position of the container?

**Comment by flackr** [comment]
I agree with @johannesodland. I think we should try to use terminology that either conveys [initial scroll position](https://www.w3.org/TR/css-overflow-3/#initial-scroll-position) or something from the html spec [try to scroll to the fragment](https://html.spec.whatwg.org/#try-to-scroll-to-the-fragment).

**Comment by DavMila** [comment]
> Wouldn't `scroll-origin-*` possibly be confusing, as we're not affecting the [scroll origin](https://www.w3.org/TR/css-overflow-3/#scroll-origin), but the initial scroll position of the container?

I also agree, but I felt since [scroll origin](https://www.w3.org/TR/css-overflow-3/#scroll-origin) is more of a spec detail rather than an API exposed to developers, it wouldn't be much of a concern. But maybe some future property might use "scroll-origin-*" in a way that aligns with [scroll origin](https://www.w3.org/TR/css-overflow-3/#scroll-origin) so it would be better to avoid that potential source of confusion if we can.

I've also thought of `initial-scroll-target` (it just sounds better than `scroll-initial-target` :) ) but I felt it might sound like it's specifying what the affected element should scroll to rather than that it should be scrolled to by its scroll container (though the same can be said of `scroll-start-target`) so I wondered if `scroll-container-target` might be a better name. It doesn't convey [initial scroll position](https://www.w3.org/TR/css-overflow-3/#initial-scroll-position) quite as directly but it might work well if we combine it with options that reflect \<scope> and \<timing> like I mention above. E.g.:
```
.target {
    scroll-container-target: local always;
}
```

To be clear I'm not proposing we adopt all these names right away, but that `scroll-container-target` is a name we can use now which would work well with other options we might think about in the future.

**Comment by DavMila** [comment]
Thinking about this a little more, perhaps `initial-scroll-target` best represents what we are trying to express here.

Firstly for the property name, I think options we should consider are:

- scroll-container-target, and
- initial-scroll-target,

between which my preference is for `initial-scroll-target` because it more directly expresses that the affected element should be scrolled to when its scroll container is first displayed than `scroll-container-target` does.

Then, for the property values, I think we should consider:

- `none` | `local`,
- `none` | `nearest`, and
- `none` | `include`,

among which my preference would be `none` | `local` because I think it would work nicely with a future expansion of the options to include `global`.

**Comment by argyleink** [comment]
I like `initial-scroll-target`, nice! Value proposals work for me.

**Comment by fantasai** [comment]
Related: https://github.com/w3c/csswg-drafts/issues/10913

**Comment by fantasai** [comment]
I think the interesting question is, what happens if you set this property via the CSSOM?

**Comment by flackr** [comment]
> I think the interesting question is, what happens if you set this property via the CSSOM?

You mean, if you call `someDescendant.style.setProperty("initial-scroll-target", "local");`?

If parsing is not finished yet this would result in it finding and scrolling to the element.
If parsing is finished this would do nothing until the box is reattached, right?

I suppose there is an argument to say until the user does some scrolling in the box it perhaps should result in updating the scroller since it still doesn't have a user indicated position. Do we do this for changes to `align-content`?
